lib/rubocop/cop/rspec/identical_equality_assertion.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for equality assertions with identical expressions on both sides.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo.bar).to eq(foo.bar)
+      #   expect(foo.bar).to eql(foo.bar)
+      #
+      #   # good
+      #   expect(foo.bar).to eq(2)
+      #   expect(foo.bar).to eql(2)
+      #
+      class IdenticalEqualityAssertion < Base
+        MSG = 'Identical equality can never fail.'
+
+        def_node_matcher :equality_check?, <<~PATTERN
+          (send (send nil? :expect $_) :to (send nil? {:eql :eq} $_))
+        PATTERN
+
+        def on_send(node)
```

Comment:
Does it make sense to use the `RESTRICT_ON_SEND` trick here?

Reply:
:100: